ckanext/who_afro/tests/test_action_dataset_tag_patch.py

```diff
@@ -43,7 +45,11 @@ def create_dataset(tags_list, dataset_name):
         id="test-id" + dataset_name,
         owner_org=org['id'],
         tags=tags,
-        name=dataset_name
+        name=dataset_name,
+        extras=[
+            {'key': 'programme', 'value': json.dumps('programme')},
```

Comment:
Something looks a little fishy here. :thinking: 

(1) `json.dumps` is doing very little.  Just turning 'programme' into '"programme"' .  Not sure it is necessary?
(2) But a bigger issue, I don't fully understand why it expects / requires the programme and country fields to be provided? It is suggesting to me that we have a poorly written code elsewhere in the code base.  Can you clarify which part of our code is requiring these values to be provided?

Reply:
@fjelltopp/who-afro not specifying `country` and `program` will cause this function in plugin to fail:
`    def before_dataset_index(self, data_dict):
        data_dict['programme'] = json.loads(data_dict['programme'])
        data_dict['country'] = json.loads(data_dict['country'])
        return data_dict`

Reply:
I think the json.loads there implies the json.dumps in the create_dataset action, consequently by removing the json.dumps() will also produce an error.

Reply:
@A-Souhei So I think it is this "before_dataset_index" function that is really the problem. I'm not sure it should be there at all.  It is also breaking things for other parts of the project. You added the function as part of the UI work.  Can you remember why?